src/shared/database/amendments.ts

```diff
@@ -0,0 +1,141 @@
+import axios from 'axios'
+import createHash from 'create-hash'
+
+import { AmendmentsInfo } from '../types'
+import logger from '../utils/logger'
+
+import { query } from './utils'
+
+const log = logger({ name: 'amendments' })
+
+const amendmentIDs = new Map<string, { name: string; deprecated: boolean }>()
+const rippledVersions = new Map<string, string>()
+
+const ACTIVE_AMENDMENT_REGEX =
+  /^\s*REGISTER_F[A-Z]+\s*\((?<amendmentName>\S+),\s*.*$/u
+const RETIRED_AMENDMENT_REGEX =
+  /^ .*retireFeature\("(?<amendmentName>\S+)"\)[,;].*$/u
+
+const AMENDMENT_VERSION_REGEX =
+  /\| \[(?<amendmentName>[a-zA-Z0-9_]+)\][^\n]+\| (?<version>v[0-9]*\.[0-9]*\.[0-9]*|TBD) *\|/u
+
+// TODO: Clean this up when this PR is merged:
+// https://github.com/XRPLF/rippled/pull/4781
```

Comment:
Please also create an issue in the repo.

Reply:
Created an issue:
https://github.com/ripple/validator-history-service/issues/162